libpromises/expand.c

```diff
@@ -632,7 +632,7 @@ void BundleResolvePromiseType(EvalContext *ctx, const Bundle *bundle, const char
             for (size_t i = 0; i < SeqLength(pt->promises); i++)
             {
                 Promise *pp = SeqAt(pt->promises, i);
-#if 1
+#if 0
```

Comment:
FWIW going the 2nd path also fixes a soft fail test, `01_vars/01_basic/unresolved_variables.cf`.

See [here](http://jenkins.usdc.cfengine.com/view/Pull%20requests/job/testing-enterprise-pr_core/502/label=PACKAGES_x86_64_linux_redhat_7/testReport/junit/._01_vars_01_basic_unresolved_variables/cf/unresolved_variables_cf/)